app/services/school_data_importer.rb

```diff
@@ -46,77 +41,120 @@ def school_data_files
+        if new_school?
+          link_school_to_local_authority
+          link_school_to_local_district_authority
+        end
+      end
     end
 
-    local_authority.name = row_local_authority_name
-    local_authority.save!
-    local_authority
-  end
+  private
+
+    def new_school?
+      @new_school
+    end
 
-  def row_to_lad(row)
-    local_authority_district = LocalAuthorityDistrict.find_or_initialize_by(code: row.fetch("DistrictAdministrative (code)"))
-    row_lad_name = row.fetch("DistrictAdministrative (name)")
+    def record_school_details
+      school.name = data.fetch("EstablishmentName")
+      school.address_line1 = data.fetch("Street")
+      school.address_line2 = data.fetch("Locality")
+      school.address_line3 = data.fetch("Town")
+      school.postcode = data.fetch("Postcode")
+      school.school_phase_type = data.fetch("PhaseOfEducation (code)").to_i
+      school.school_phase_name = data.fetch("PhaseOfEducation (name)")
+      school.school_website = data.fetch("SchoolWebsite")
+      school.primary_contact_email = data.fetch("MainEmail").presence
+      school.secondary_contact_email = data.fetch("AlternativeEmail").presence
 
-    if local_authority_district.persisted? && local_authority_district.name != row_lad_name
-      logger.info "LA name change in school import. Old name: #{local_authority_district.name}, New name: #{row_lad_name}"
+      # Changes to properties below carries major consequences and must be avoided until we know how to handle them
+      if new_school?
+        school.ukprn = data.fetch("UKPRN")
+
+        school.school_type_code = data.fetch("TypeOfEstablishment (code)").to_i
+        school.school_type_name = data.fetch("TypeOfEstablishment (name)")
+
+        school.school_status_code = data.fetch("EstablishmentStatus (code)").to_i
+        school.school_status_name = data.fetch("EstablishmentStatus (name)")
+
+        school.administrative_district_code = data.fetch("DistrictAdministrative (code)")
+        school.administrative_district_name = data.fetch("DistrictAdministrative (name)")
+      end
+
+      school.save!
     end
 
-    local_authority_district.name = row_lad_name
-    local_authority_district.save!
-    local_authority_district
-  end
+    def local_authority
+      return @local_authority if @local_authority
+
+      code = data.fetch("LA (code)")
+      @la_cache.fetch(code) do
+        local_authority = LocalAuthority.find_or_initialize_by(code: code)
+        new_name = data.fetch("LA (name)")
 
-  def link_school_to_local_authority(school, local_authority)
-    school_local_authority = SchoolLocalAuthority.find_or_initialize_by(school: school, local_authority: local_authority)
+        if local_authority.name != new_name
+          if local_authority.persisted?
+            logger.info "LA name changed in school import. Old name: #{local_authority.name}, New name: #{new_name}"
+          end
```

Comment:
The `else` here is that GIAS is inconsistent in it's naming, right? I think this is interesting enough to log

Reply:
That's a bit where it is a bit shaky - the else is when we have no local authority in the database at all (in which case name is `nil`). This code is assuming that the CSV file we recceive is internally correct, so it does not contain two different names for the same LA codes) to such an extend that the name is only checked on the first encounter of LA code in the file - each next encounter of the same code will fetch LA record from the provided cache (`fetch` only executes the block when given key is not present).

Reply:
Understood, thanks